### PR TITLE
2354 fix soft link sizes

### LINKED ISSUE #2354: Can't download or stream untranscoded media if the media store is a symlinked file
**Media Info of the file**
Not relevant, it is however a symlink to the real file which is in another location, both of which are accessible to the Jellyfin service

**Logs**
```
Jan 30 19:00:34 avian jellyfin[23804]: [19:00:34] [INF] User policy for mike. EnablePlaybackRemuxing: True EnableVideoPlaybackTranscoding: True EnableAudioPlaybackTranscoding: True
Jan 30 19:00:34 avian jellyfin[23804]: [19:00:34] [INF] Profile: Unknown Profile, Path: /FILENAME/REDACTED.mkv, isEligibleForDirectPlay: True, isEligibleForDirectStream: True
Jan 30 19:00:34 avian jellyfin[23804]: [19:00:34] [ERR] Error processing request
Jan 30 19:00:34 avian jellyfin[23804]: System.InvalidOperationException: Response Content-Length mismatch: too many bytes written (131072 of 138).
Jan 30 19:00:34 avian jellyfin[23804]:    at Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http.HttpProtocol.ThrowTooManyBytesWritten(Int32 count)
Jan 30 19:00:34 avian jellyfin[23804]:    at Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http.HttpProtocol.VerifyAndUpdateWrite(Int32 count)
Jan 30 19:00:34 avian jellyfin[23804]:    at Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http.HttpProtocol.InitializeResponseAsync(Int32 firstWriteByteCount)
Jan 30 19:00:34 avian jellyfin[23804]:    at Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http.HttpProtocol.WriteAsync(ReadOnlyMemory'1 data, CancellationToken cancellationToken)
Jan 30 19:00:34 avian jellyfin[23804]:    at Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http.HttpResponseStream.WriteAsync(Byte[] buffer, Int32 offset, Int32 count, CancellationToken cancellationToken)
Jan 30 19:00:34 avian jellyfin[23804]:    at Microsoft.AspNetCore.ResponseCompression.BodyWrapperStream.WriteAsync(Byte[] buffer, Int32 offset, Int32 count, CancellationToken cancellationToken)
Jan 30 19:00:34 avian jellyfin[23804]:    at System.IO.Stream.CopyToAsyncInternal(Stream destination, Int32 bufferSize, CancellationToken cancellationToken)
Jan 30 19:00:34 avian jellyfin[23804]:    at Emby.Server.Implementations.HttpServer.FileWriter.TransmitFile(Stream stream, String path, Int64 offset, Int64 count, FileShareMode fileShareMode, CancellationToken cancellationToken)
Jan 30 19:00:34 avian jellyfin[23804]:    at Emby.Server.Implementations.HttpServer.FileWriter.WriteToAsync(HttpResponse response, CancellationToken cancellationToken)
Jan 30 19:00:34 avian jellyfin[23804]:    at Emby.Server.Implementations.Services.ServiceHandler.ProcessRequestAsync(HttpListenerHost httpHost, IRequest httpReq, HttpResponse httpRes, ILogger logger, CancellationToken cancellationToken)
Jan 30 19:00:34 avian jellyfin[23804]:    at Emby.Server.Implementations.HttpServer.HttpListenerHost.RequestHandler(IHttpRequest httpReq, String urlString, String host, String localPath, CancellationToken cancellationToken)
```

Note the '138' number changes depending on the media file being loaded, I have determined that is in fact the path the symlink is pointing to.
Example: a symlink to "../../Media/foobar.baz" would be 22 while "../../Media/blah-blah-blah.blah" would be 31

**FFmpeg Logs**
Not relevant, transcoding isn't happening during failure

**Stats for Nerds Screenshots**
Not available in the places this problem is reproducable

**Server System (please complete the following information):**
 - OS: Linux/Debian 10
 - Jellyfin Version: 10.4.2
 - Hardware settings & device: VM running headless so I've not enabled any hardware decoding
 - Reverse proxy: Apache
 - Other hardware notes: Media is mounted via NFS from the host OS
 
**Client System (please complete the following information):**
 - Device: Xiaomi Redmi Note 5 Pro, & my low-end gaming desktop (Nvidia GPU)
 - OS: Android LineageOS 15.1, & Debian 10
 - Client: org.jellyfin.mobile, Chromecast, jellyfin-mpv-shim, and Google Chrome
 - Browser (if Web client): Google Chrome when relevant.
 - Client and Browser Version: org.jellyfin.mobile 0.9.7 & Google Chrome 79.0.3945.117 (Official Build) (64-bit)

### PR DESCRIPTION
**Pull request by Ken-g6**
**Changes**
Open files that appear to be symlinks for reading, to read the file size.

**Issues**
Fixes #2354

### CODE REVIEW

**Comment by Ken-g6** [comment]
Note: It has been suggested to use .FullName instead of .toString().  I believe I tried that, my compiler said .FullName was a private property, and .toString() was the suggested fix I found online.

File: Emby.Server.Implementations/IO/ManagedFileSystem.cs [hunk 1]
```diff
@@ -245,6 +245,16 @@ private FileSystemMetadata GetFileSystemMetadata(FileSystemInfo info)
                 if (info is FileInfo fileInfo)
                 {
                     result.Length = fileInfo.Length;
+
+                    // Issue #2354 get the size of files behind symbolic links
+                    if(fileInfo.Attributes.HasFlag(FileAttributes.ReparsePoint))
```
Comment (crobibero) [inline Emby.Server.Implementations/IO/ManagedFileSystem.cs:250]:
```suggestion
                    if (fileInfo.Attributes.HasFlag(FileAttributes.ReparsePoint))
```

Current spacing should be throwing a warning

File: Emby.Server.Implementations/IO/ManagedFileSystem.cs [hunk 2]
```diff
@@ -245,6 +245,16 @@ private FileSystemMetadata GetFileSystemMetadata(FileSystemInfo info)
...
+                    {
+                        using (Stream thisFileStream = File.OpenRead(fileInfo.ToString()))
```
Comment (crobibero) [inline Emby.Server.Implementations/IO/ManagedFileSystem.cs:252]:
```suggestion
                        using (Stream thisFileStream = File.OpenRead(fileInfo.FullName))
```

`FullName` works in my environment

File: Emby.Server.Implementations/IO/ManagedFileSystem.cs (hunk 1 above)
Comment (Ken-g6) [reply Emby.Server.Implementations/IO/ManagedFileSystem.cs]: Is there a coding standard for Jellyfin?  I didn't see one.

Comment (crobibero) [reply Emby.Server.Implementations/IO/ManagedFileSystem.cs]: We use stylecop in jellyfin-server. It should have given you a warning when compiling or editing the code depending on your IDE settings

**Pull request merged**